posts/modular-monolith-microservices.md

```diff
@@ -0,0 +1,76 @@
+
+Organizations and teams worldwide have approached application development with an agile mindset, hardening the process of developing a monolith. Making a single change to your application's server or client-side can become an extensive process, with having to roll out the entire application again. This resource-intensive activity can make CI/CD difficult, so application architects and developers have leaned into a different type of architecture - microservice architecture. We'll first take a look at the Modular Monolith and how it differs from a traditional monolith before moving into microservices.
+
+![Monolith Vs Microservice Architecture Diagram](../images/2022/12/mono-micro-diagram.png) ***Microservice vs Monolith Architecture***
+
+## Modular Monolith
+
+A Modular Monolith consists of dividing logic into modules, as each module is independent and isolated. Every module is in charge of its own business logic, database or schema, etc. With this model of architecture, you can build and modify existing layers of each module while leaving the others unaffected.
+
+Modules should communicate through APIs, allowing access to the logic of each module through public methods, not internal functions & the logic of each one. This architecture model is less complex than microservices and more scalable and easier to maintain than traditional monolithic apps. 
+
+One of the notable disadvantages for both monoliths and modular monoliths is how slow the development velocity is. Adding new features is slow because there is a large cognitive load on the developer. Modules of monoliths are tightly coupled, which can significantly lengthen the release cycle of a monolith.
+
+While this architecture model has its advantages over a monolithic application, it proves to still be difficult to modernize. It takes a lot of time since the velocity for this model is slow and the release cycle is sluggish. This is something companies need to consider as many applications need to be modernized for several reasons (i.e. modern hardware, browser updates, network bandwidth, tech stack, etc.)
+
+![Monolith Vs Microservice Architecture Diagram](../images/2022/12/modular-monolith.png) ***Modular Monolith Architecture***
+
+## Microservices
+
+No longer considered a new architectural model, microservices have proven to be successful and adopted in approximately [63%](https://dzone.com/articles/new-research-shows-63-percent-of-enterprises-are-a) of enterprises. Most companies who have converted have mentioned that their new architecture has improved customer & end-to-end experience and cut costs on infrastructure and other development tools. 
+
+It is essential to recognize that breaking down a monolith and developing microservices can be expensive and time-consuming. Understanding the benefits of transitioning to microservices and why other organizations are doing the same is vital. Some reasons companies should leverage microservices may include the following: 
```

Comment:
the sentence where you say "It is essential to recognize that breaking down a monolith and developing microservices can be expensive and time-consuming." sort of sounds like you're down-playing microservices but the following sentences sound like you're saying it's a good idea to leverage microservices